src/models/userDao.js

```diff
@@ -0,0 +1,179 @@
+const { AppDataSource } = require("../models/dataSource");
+const { createConnection } = require("typeorm");
+const { throwError } = require("../utils");
+
+// isSubscribed 추가
+const findByKakaoId = async (kakaoId) => {
+  const [result] = await AppDataSource.query(
+    `
+        SELECT id, user_type
+        FROM users
+        WHERE kakao_id = ?
+        `,
+    [kakaoId]
+  );
+
+  console.log("userDao findByKakaoId result: ", result);
+
+  return result;
+
+};
+
+const updateImgUrl = async (userId, imgUrl) => {
+  const result = await AppDataSource.query(
+    `
+      UPDATE users 
+      SET img_url = ?
+      WHERE id = ?
+    `,
+    [imgUrl, userId]
+  );
+
+  return result;
+};
+
+const createUser = async (kakaoId, imgUrl) => {
+  const result = await AppDataSource.query(
+    `
+    INSERT INTO users 
+    (kakao_id, img_url) 
+    VALUES (?, ?)
+    `,
+    [kakaoId, imgUrl]
+  );
+  if (result.insertId) {
+    
+    return {
+      id: result.insertId,
+      //img_url: imgUrl,
+    };
+
+  } else {
+    throwError(401, "FAIL_TO_CREATE_USER")
+  }
+
+};
+
+const isSubscribed = async (userId) => {
+  const [result] = await AppDataSource.query(
+    `
+      SELECT *
+      FROM sub_orders
+      WHERE user_id = ?
+      AND end_at > NOW()
+    `,
+    [userId]
+  )
```

Comment:
```suggestion
  const [result] = await AppDataSource.query(
    `
      SELECT id
      FROM sub_orders
      WHERE user_id = ?
      AND end_at > NOW()
      LIMIT 1
    `,
    [userId]
  )
```
- `*`로 모든 데이터를 다 가져오는 과정은 지양하는게 좋습니다. 존재 여부만 판단하니 id만 있으면 될 것같아요.
- 기본적으로 SELECT + WHERE 문의 경우 데이터의 전체 갯수를 다 하나하나 점검합니다. 따라서 있는지 여부만 확인하는 컬럼이라면 LIMIT 1을 걸어서, 하나 찾자마자 곧바로 탐색을 멈추게 하여 데이터베이스의 검색 과정을 낭비하지 않을 수 있습니다.
- 애초에 테이블을 생성할 때부터 WHERE 절에 있는 `user_id`를 unique key로 걸어두었다면, 그 조건만으로도 충분하긴 합니다. 아마 unique key constraint 없을 것 같아서 리뷰드립니다!